src/Features/Core/Portable/Completion/Providers/TypeImportCompletionItem.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using System.Globalization;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.Completion.Providers
+{
+    internal static class TypeImportCompletionItem
+    {
+        private const string GenericTypeNameManglingString = "`";
+        private static readonly string[] s_IntegerOneToNine = { "1", "2", "3", "4", "5", "6", "7", "8", "9" };
+        private static readonly string[] s_aritySuffixesOneToNine = { "`1", "`2", "`3", "`4", "`5", "`6", "`7", "`8", "`9" };
```

Comment:
minor point.  it seems weird that you're taking an arity (an int), and converting it to a string to put in a dictionary.  then you get the sstring out, and get an int again, only to get the arity suffix.  Why not just simply store the arity suffix in the dictionary?

Reply:
Good catch.